contributors/design-proposals/pod-injection-policy.md

```diff
@@ -0,0 +1,676 @@
+  injected.
+
+> **Note:** In the case of a conflict nothing will be injected. The entire
+> policy is ignored and an event is thrown on the pod detailing the conflict.
+
+### AdmissionControl Plug-in: PodInjectionPolicy
+
+The **PodInjectionPolicy** plug-in introspects all incoming pod creation
+requests and injects the pod based off a `Selector` with the desired
+attributes.
+
+For the initial alpha, the order of precedence for applying multiple
+`PodInjectionPolicy` specs is from oldest to newest. All Pod Injection
+Policies in a namespace should be order agnostic. However we can use merge keys
+to detect some of the conflicts that may occur.
+
+This will not be enabled by default for all clusters, but once GA will be
+a part of the set of strongly recommended plug-ins documented
+[here](https://kubernetes.io/docs/admin/admission-controllers/#is-there-a-recommended-set-of-plug-ins-to-use).
+
+**Why not an Initializer?**
+
+This will be first implemented as an AdmissionControl plug-in then can be
+converted to an Initializer once that is fully ready. The proposal for
+Initializers can be found at [kubernetes/community#132](https://github.com/kubernetes/community/pull/132).
+
+
+#### Behavior
+
+This will modify the pod spec. The supported changes to
+`Env`, `EnvFrom`, and `VolumeMounts` apply to the container spec for
+all containers in the pod with the specified matching `Selector`. The
+changes to `Volumes` apply to the pod spec for all pods matching `Selector`.
+
+The resultant modified pod spec will be annotated to show that it was modified by
+the `PodInjectionPolicy`. This will be of the form
+`podinjectionpolicy.admission.kubernetes.io/<pip name>": "<resource version>"`.
+
+*Why modify all containers in a pod?*
+
+Currently there is no concept of labels on specific containers in a pod which
+would be necessary for per-container pod injections. We could add labels
+for specific containers which would allow this and be the best solution to not
+injecting all. Container labels have been discussed various times through
+multiple issues and proposals, which all congregate to this thread on the
+[kubernetes-sig-node mailing
+list](https://groups.google.com/forum/#!topic/kubernetes-sig-node/gijxbYC7HT8).
+In the future, even if container labels were added, we would need to be careful
+about not making breaking changes to the current behavior.
+
+Other solutions include basing the container to inject based off
+matching its name to another field in the `PodInjectionPolicy` spec, but
+this would not scale well and would cause annoyance with configuration
+management.
+
+In the future we might question whether we need or want containers to express
+that they expect injection. At this time we are deferring this issue.
+
+#### Optional Behavior with implicit injections
+
+The above described the behavior for explicitly defining what should be
+injected into a pod.
+
+A controller has been discussed to automatically fill in the
```

Comment:
Has a decision been made regarding implicit vs explicit? Or at least a summary of the pros and cons?